<launch>
  <!-- Arguments -->
  <arg name="robot_name" default="thorvald_001"/>
  <arg name="map_file" default="$(find uol_cmp9767m_base)maps/cropped.yaml"/>
  <arg name="fake_loc" default="false"/>

  <!-- Simulation -->
  <include file="$(find uol_cmp9767m_base)/launch/thorvald-sim.launch">
    <arg name="obstacles" default="true" />
    <arg name="second_robot" default="false" />
    <arg name="fake_localisation" value= "$(arg fake_loc)"/>
  </include>

  <!-- Localisation -->
  <include file="$(find uol_cmp9767m_tutorial)/launch/amcl.launch" if="$(eval fake_loc=='false')"/>
  <node name="map_server" pkg="map_server" type="map_server" args="$(arg map_file)" if="$(eval fake_loc=='false')"/>

  <!-- move_base -->
  <node name="move_base" pkg="move_base" type="move_base" output="screen">
        <remap from="odom" to="$(arg robot_name)/odometry/base_raw" />
        <remap from="cmd_vel" to="$(arg robot_name)/nav_vel" />

        <rosparam file="$(find uol_cmp9767m_tutorial)/config/costmap_common_params.yaml" command="load" ns="global_costmap"/>
        <rosparam file="$(find uol_cmp9767m_tutorial)/config/costmap_common_params.yaml" command="load" ns="local_costmap"/>

        <rosparam file="$(find uol_cmp9767m_tutorial)/config/global_costmap_params.yaml" command="load"/>
        <rosparam file="$(find uol_cmp9767m_tutorial)/config/local_costmap_params.yaml" command="load"/>

        <rosparam file="$(find uol_cmp9767m_tutorial)/config/planners.yaml" command="load"/>
  </node>
  <node type="rviz" name="rviz" pkg="rviz" args="-d $(find rp_package)/rviz/rp_config.rviz" />
</launch>
